x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -723,32 +740,54 @@ public void testSnapshotMountedIndexWithTimestampsRecordsTimestampRangeInIndexMe
         mountSnapshot(repositoryName, snapshotOne.getName(), indexName, indexName, Settings.EMPTY);
         ensureGreen(indexName);
 
-        final IndexLongFieldRange timestampRange = clusterAdmin().prepareState()
+        final IndexMetadata indexMetadata = clusterAdmin().prepareState()
             .clear()
             .setMetadata(true)
             .setIndices(indexName)
             .get()
             .getState()
             .metadata()
-            .index(indexName)
-            .getTimestampRange();
+            .index(indexName);
 
+        final IndexLongFieldRange timestampRange = indexMetadata.getTimestampRange();
         assertTrue(timestampRange.isComplete());
 
+        final IndexLongFieldRange eventIngestedRange = indexMetadata.getEventIngestedRange();
+        assertTrue(eventIngestedRange.isComplete());
+
         if (indexed) {
             assertThat(timestampRange, not(sameInstance(IndexLongFieldRange.UNKNOWN)));
+            assertThat(eventIngestedRange, not(sameInstance(IndexLongFieldRange.UNKNOWN)));
             if (docCount == 0) {
                 assertThat(timestampRange, sameInstance(IndexLongFieldRange.EMPTY));
+                assertThat(eventIngestedRange, sameInstance(IndexLongFieldRange.EMPTY));
             } else {
                 assertThat(timestampRange, not(sameInstance(IndexLongFieldRange.EMPTY)));
-                DateFieldMapper.Resolution resolution = dateType.equals("date")
+                DateFieldMapper.Resolution timestampResolution = dateType.equals("date")
```

Comment:
This rename seems unnecessary and I think we can use the same `resolution` for validating both timestamp ranges, since we use the same `dateType` anyway?

Reply:
I don't think so? My understanding is that resolution can be different between the two - one could be millis resolution and the other could be nanos. Or is there some invariant enforced in the production code somewhere that both @timestamp and `event.ingested` have to use the same resolution?

Reply:
It is more to do with the test code. [Here](https://github.com/elastic/elasticsearch/pull/106252/files/4dcfea5dfb5dda3e6af760a957f7975570833c50#diff-f1bb44f1e1546df7098770be6ad7127aea7fa2ff3e7d4601e90e4d3ad520944fR673) we pick a single one of ("date", "date_nanos"), used for both fields in this test. Thus the expected resolution of both fields will also be the same due to how the test is written.